### ISSUE TITLE
The ResampleSpecStep and Extract1DStep within MasterBackgroundMosStep are inaccessible

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#8429]
_Issue [JP-3598](https://jira.stsci.edu/browse/JP-3598) was created on JIRA by [Kayli Glidic](https://jira.stsci.edu/secure/ViewProfile.jspa?name=kglidic):_

In calwebb_spec2, the MasterBackgroundMosStep is responsible for handling master background subtraction with MOS data. While users can readily modify parameters for calibration steps like barshadow, flat_field, pathloss, and photom within the MasterBackgroundMosStep, there are constraints when it comes to accessing and modifying the ResampleSpecStep and Extract1dStep, which appear to be hard-coded. These two should be accessible like the other calibration steps within the MasterBackgroundMosStep.

 

For instance, users may find it beneficial to adjust parameters such as weight_type or fillval in the ResampleSpecStep prior to the extraction of 1-D background spectra. Additionally, users may wish to modify the background extraction region by overriding the default reference file, a capability currently unavailable. Just as an example, if users observe background with a 3-shutter-slitlet, by default the pipeline will only extract a 6-pixel wide region from the central shutter. However, there may be valid background in those adjacent shutters which users may wish to also include in the extraction if they could modify the reference file.

 

We would like the ability to define parameters for the ResampleSpecStep and Extract1dStep in a manner similar to how we can for barshadow within the 'steps' section under 'master_background_mos', as in this example:

 

 
```java
spec2_params = {
  'master_background_mos': {
    'skip': False,
    'save_background': True,
    'steps': {
      'barshadow': {'override_barshadow': file_path},
    }
  }
}
 
Spec2Pipeline.call(rate_file, save_results=True, output_dir=output_dir, steps=spec2_params)
```

**Comment by stscijgbot-jp** [comment]
_Comment by [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) on [JIRA](https://jira.stsci.edu/browse/JP-3598?focusedCommentId=702214#comment-702214):_

[Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) Any thoughts on this?

**Comment by stscijgbot-jp** [comment]
_Comment by [Melanie Clarke](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mclarke) on [JIRA](https://jira.stsci.edu/browse/JP-3598?focusedCommentId=759287#comment-759287):_

Fixed by #8847

**Comment by stscijgbot-jp** [comment]
_Comment by [Christian Hayes](https://jira.stsci.edu/secure/ViewProfile.jspa?name=chayes) on [JIRA](https://jira.stsci.edu/browse/JP-3598?focusedCommentId=942486#comment-942486):_

[Kayli Glidic](https://jira.stsci.edu/secure/ViewProfile.jspa?name=kglidic) I went ahead and tested this while testing the addition of pixel_replace to the MOS master background.  These step parameters are now accessible.  Closing.
